internal/dsync/dsync_test.go

```diff
@@ -236,10 +236,38 @@ func TestTwoSimultaneousLocksForDifferentResources(t *testing.T) {
 	time.Sleep(10 * time.Millisecond)
 }
 
-// Test refreshing lock
+// Test refreshing lock - refresh should always return true
+//
+func TestSuccessfulLockRefresh(t *testing.T) {
+	dm := NewDRWMutex(ds, "aap")
+	contextCanceled := make(chan struct{})
+
+	ctx, cl := context.WithCancel(context.Background())
+	cancel := func() {
+		cl()
+		close(contextCanceled)
+	}
+
+	if !dm.GetLock(ctx, cancel, id, source, Options{Timeout: 5 * time.Minute}) {
+		t.Fatal("GetLock() should be successful")
+	}
+
+	timer := time.NewTimer(drwMutexRefreshInterval * 2)
```

Comment:
Does this test take 20 seconds to run?

Reply:
> Does this test take 20 seconds to run?

This test is checking that a lock is not canceled accidently. I am giving it two rounds of Refresh before we can be sure it is is not buggy.

Reply:
@vadmeste My point is that no test should take 20 seconds to run.

Reply:
> @vadmeste My point is that no test should take 20 seconds to run.

Refresh is executed each 10 seconds and I think we don't have a choice here. I can skip it when -short flag is passed

Reply:
> I can skip it when -short flag is passed

If you don't feel like refactoring, then sure. And make it `t.Parallel`.

Reply:
t.Parallel will need refactoring because tests share locking API and there is some weird RPC code which looks like legacy. I will send another PR with initialize a new locking API for each test so we can run then in parallel. That will reduce test duration in this package from 2 minutes to 20 seconds.